### DISCUSSION TITLE
Alternative Detectors

### DISCUSSION THREAD

**Discussion by Salvora** [blakeblackshear/frigate#15637] [Ask A Question] [answered]
What do you guys think about the alternative detectors other than Coral? Like this one?
https://www.tomshardware.com/tech-industry/artificial-intelligence/memryx-launches-usd149-mx3-m-2-ai-accelerator-module-capable-of-24-tops-compute-power
"offering a total of up to 24 TOPS of compute power while consuming just 6 to 8 watts of power"

**Comment by NickM-27** [comment]
Frigate supports many detectors https://docs.frigate.video/configuration/object_detectors

And in 0.15 support for Hailo and AMD ROCm is added as well

**Reply by Salvora** [reply]
I guess I should rephrase my question. Alternative detectors similar to Coral? I just saw the news on this one from MemryX. Specs seem much better than the Coral. Coral is now what, 5 years old?

**Reply by NickM-27** [reply, marked as answer]
That would be the Hailo https://hailo.ai/products/ai-accelerators/hailo-8-ai-accelerator/

**Reply by Salvora** [reply]
First time heard about that but looking at the specs, it also seems very good. But frigate does not support it according to docs.

**Reply by NickM-27** [reply]
Like I said it's supported in 0.15 https://deploy-preview-13787--frigate-docs.netlify.app/configuration/object_detectors

**Reply by Salvora** [reply]
Excellent, and I find it amazing that this project follows the latest and greatest. Though it is not related, just wanted to ask; since we will be getting all kinds of new accelerators from different brands in the future due to AI boom going on, how likely is it to make the detector integration to frigate plug-gable with add-ons? I know it is an open source project and there are "pull requests" and options to fork your own, however, I believe it is not really easy to integrate a completely new detector without breaking other stuff.
The idea is to create an environment where it is easier to create detector addons for frigate to load (like what HA does this with HACS and custom integrations)

**Reply by NickM-27** [reply]
I think your understanding is incorrect. The detectors in Frigate are already "plugin" format meaning adding support to Frigate itself is as easy as adding one file. However, detectors almost always require additional installation into the docker image itself  to interface with the hardware which is where things are more complicated and not as simple as just installing python code like HACS works. 

For this purpose Frigate has the [community supported boards framework](https://docs.frigate.video/development/contributing-boards) where users of the community can create and contribute detectors.
